### PR TITLE
Add vectorized_particles to ELBO

### PR DESCRIPTION
**Pull request by fehiepsi**
Allow to use lax.map instead of vmap in ELBO to reduce memory requirement

### CODE REVIEW

File: numpyro/infer/elbo.py [hunk 1]
```diff
@@ -33,6 +34,8 @@ class ELBO:
 
     :param num_particles: The number of particles/samples used to form the ELBO
         (gradient) estimators.
+    :param vectorize_particles: Whether to use `jax.vmap` to obtain elbos over the
```
Comment (martinjankowiak) [inline numpyro/infer/elbo.py:37]: is this only for eval or also for training?

Comment (fehiepsi) [reply numpyro/infer/elbo.py:37]: this is for both.

Comment (martinjankowiak) [reply numpyro/infer/elbo.py:37]: nit:
> :param vectorize_particles: Whether to use `jax.vmap` to compute ELBOs over the `num_particles`-many particles in parallel. If False use `jax.lax.map`. Defaults to True.

File: numpyro/infer/elbo.py [hunk 2]
```diff
@@ -108,11 +112,10 @@ class Trace_ELBO(ELBO):
...
```
Comment (martinjankowiak) [inline numpyro/infer/elbo.py:115]: ditto

File: numpyro/infer/elbo.py [hunk 3]
```diff
@@ -311,6 +320,8 @@ class RenyiELBO(ELBO):
         Here :math:`\alpha \neq 1`. Default is 0.
     :param num_particles: The number of particles/samples
         used to form the objective (gradient) estimator. Default is 2.
+    :param vectorize_particles: Whether to use `jax.vmap` to obtain elbos over the
+        particles. If False, we will use `jax.lax.map`. Defaults to True.
```
Comment (martinjankowiak) [inline numpyro/infer/elbo.py:324]: ditto

**Review by martinjankowiak — APPROVED** [review]
lgtm

**Pull request merged**